private/bufpkg/bufmodule/targeting_module.go

```diff
@@ -148,13 +213,53 @@ func (m *targetingModule) TargetFileInfos(ctx context.Context) (fileInfos []bufm
 	// if !allowNotExist, i.e. if all targetPaths must have a matching file,
 	// we check the matchingPotentialDirPathMap against the potentialDirPathMap
 	// to make sure that potentialDirPathMap is covered
-	if !m.targetPathsAllowNotExistOnWalk {
+	if !m.pathsAllowNotExistOnWalk {
 		for potentialDirPath := range potentialDirPathMap {
 			if _, ok := matchingPotentialDirPathMap[potentialDirPath]; !ok {
 				// no match, this is an error given that allowNotExist is false
 				return nil, fmt.Errorf("path %q has no matching file in the module", potentialDirPath)
 			}
 		}
+		for excludePath := range excludePathMap {
+			if _, ok := matchingExcludePaths[excludePath]; !ok {
+				// no match, this is an error given that allowNotExist is false
+				return nil, fmt.Errorf("path %q has no matching file in the module", excludePath)
+			}
+		}
 	}
 	return fileInfos, nil
 }
+
+func excludeFile(
+	fileMatchingPathMap map[string]struct{},
+	fileMatchingExcludePathMap map[string]struct{},
+) bool {
```

Comment:
It looks like we're defining the same function in two places - can we DRY this up and define this somewhere so that it can be reused?

It's only interacting with `normalpath`, so `normalpath` might be an appropriate home, but somewhere else might better since it's tailored to such a specific use case.

If it moved there, it would need to be named more generically as a filter or something else along those lines, e.g.

```go
// FilterAllEqualOrContainingPathMap returns the elements in the input that do not equal or contain a path in the filter.
func FilterAllEqualOrContainingPathMap(input map[string]struct{}, filter map[string]struct{}, pathType PathType) map[string]struct{} { ... }
```

Reply:
Yeah, the only difference here is because we have the case where `fileMatchingPathMap == 0` could be because `targetPaths == nil` (in the case where no `targetPaths` have been provided. Let me see if there is a reasonable way to clean this up and make this more generic.